src/main/java/wolox/training/services/external/OpenLibraryService.java

```diff
@@ -0,0 +1,50 @@
+package wolox.training.services.external;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Optional;
+import kong.unirest.HttpResponse;
+import kong.unirest.JsonNode;
+import kong.unirest.Unirest;
+import kong.unirest.UnirestException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.http.HttpStatus;
+import wolox.training.dtos.OpenLibraryBook;
+import wolox.training.factories.dtos.OpenLibraryBookBuilder;
+
+public class OpenLibraryService {
+
+    private final ObjectMapper mapper = new ObjectMapper();
+
+    public Optional<OpenLibraryBook> bookInfo(String isbn) {
+        String key = "ISBN:" + isbn;
+        try {
+            HttpResponse<JsonNode> response = Unirest.get("https://openlibrary.org/api/books")
+                .header("accept", "application/json")
+                .queryString("bibkeys", key)
+                .queryString("format", "json")
+                .queryString("jscmd", "data")
+                .asJson();
+            if (response.getStatus() == HttpStatus.OK.value()) {
+                JSONObject contents = response.getBody().getObject().getJSONObject(key);
+                OpenLibraryBookBuilder builder = new OpenLibraryBookBuilder();
+                builder.title(contents.getString("title"));
+                builder.subtitle(Optional.ofNullable(contents.getString("subtitle")));
+                builder.numberOfPages(contents.getInt("number_of_pages"));
+                builder.publishDate(contents.getString("publish_date"));
+                JSONArray authors = contents.getJSONArray("authors");
+                for (int i = 0; i < authors.length(); ++i) {
+                    builder.author(authors.getJSONObject(i).getString("name"));
+                }
+                JSONArray publishers = contents.getJSONArray("publishers");
+                for (int i = 0; i < publishers.length(); ++i) {
+                    builder.publisher(publishers.getJSONObject(i).getString("name"));
+                }
+                return Optional.of(builder.build());
+            }
+        } catch (UnirestException e) {
+            e.printStackTrace();
```

Comment:
Be careful with these unhandled exceptions, you should throw a custom exception showing an appropriate error message, rather than the stack trace of the original exception